### PR TITLE
Consistently use fgMakeMultiUse in the gtNewSimd*Node APIs

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves a known issue with several of the helper APIs in that they were using `impCloneExpr` even though they were not exclusive to import.

### CODE REVIEW

**Review by kunalspathak — COMMENTED** [review]
Added some questions.

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -20158,12 +20135,12 @@ GenTree* Compiler::gtNewSimdCmpOpNode(genTreeOps  op,
...
                                                    CORINFO_TYPE_INT, simdSize, isSimdAsHWIntrinsic);
-                    v = gtNewSimdHWIntrinsicNode(type, v, gtNewIconNode(SHUFFLE_ZZXX, TYP_INT), NI_SSE2_Shuffle,
-                                                 CORINFO_TYPE_INT, simdSize, isSimdAsHWIntrinsic);
                     u = gtNewSimdHWIntrinsicNode(type, u, gtNewIconNode(SHUFFLE_WWYY, TYP_INT), NI_SSE2_Shuffle,
                                                  CORINFO_TYPE_INT, simdSize, isSimdAsHWIntrinsic);
+                    v = gtNewSimdHWIntrinsicNode(type, v, gtNewIconNode(SHUFFLE_ZZXX, TYP_INT), NI_SSE2_Shuffle,
+                                                 CORINFO_TYPE_INT, simdSize, isSimdAsHWIntrinsic);
 
-                    op2 = gtNewSimdBinOpNode(GT_AND, type, v, u, simdBaseJitType, simdSize, isSimdAsHWIntrinsic);
+                    op2 = gtNewSimdBinOpNode(GT_AND, type, u, v, simdBaseJitType, simdSize, isSimdAsHWIntrinsic);
```
[the remark this answers is not in the document]
Comment (tannergooding) [reply src/coreclr/jit/gentree.cpp:20143]: Same as above. This is `GT_AND` and so is commutative. We need to order the parameters correctly to preserve evaluation order.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by tannergooding** [comment]
Test failures are known, being disabled in https://github.com/dotnet/runtime/pull/80522

**Pull request merged**